src/main/java/org/isf/permissions/manager/GroupPermissionManager.java

```diff
@@ -72,12 +77,34 @@ public GroupPermission create(UserGroup userGroup, Permission permission) throws
 		return operations.create(groupPermission);
 	}
 
+	@Transactional
+	public List<Permission> update(UserGroup userGroup, List<Integer> permissionIds, Boolean replace) throws OHDataValidationException {
+
+		var permissions = permissionOperations.findByIdIn(permissionIds).stream().toList();
+
+		var groupPermissions = operations.findUserGroupPermissions(userGroup.getCode()).stream().toList();
+
+		var permissionsToAssign = permissions.stream()
```

Comment:
for instance here `var` is used to create a list that after is passed as method parameter and this for sure must be avoided.